src/calendar.cpp

```diff
@@ -398,6 +398,14 @@ int calendar::season_length()
     return cached_season_length;
 }
 
+int calendar::turn_of_year() const {
+	return (season_turns() * season) + (turn_number % season_turns());
```

Comment:
It still has the same functionality as before, but if eternal season is enabled it will only return the turn of year corresponding to that season.
weather_generator::get_weather(...)  can then simulate temperatures for 4 equal seasons in a year.